dotnet/src/SemanticKernel.Abstractions/Memory/AIContextBehaviorsManager.cs

```diff
@@ -0,0 +1,144 @@
+    /// <returns>A task that represents the asynchronous operation, containing the combined context from all <see cref="AIContextBehavior"/> objects.</returns>
+    public async Task<AIContextPart> OnModelInvokeAsync(ICollection<ChatMessage> newMessages, CancellationToken cancellationToken = default)
+    {
+        var subContexts = await Task.WhenAll(this.Behaviors.Select(x => x.OnModelInvokeAsync(newMessages, cancellationToken)).ToList()).ConfigureAwait(false);
+        subContexts = subContexts.Where(x => x != null).ToArray();
+
+        var combinedContext = new AIContextPart();
+        combinedContext.AIFunctions = subContexts.Where(x => x.AIFunctions != null).SelectMany(x => x.AIFunctions).ToList();
+        combinedContext.Instructions = string.Join("\n", subContexts.Where(x => !string.IsNullOrWhiteSpace(x.Instructions)).Select(x => x.Instructions));
+        return combinedContext;
+    }
+
+    /// <summary>
+    /// Called when the current conversion is temporarily suspended and any state should be saved.
+    /// </summary>
+    /// <param name="threadId">The ID of the current thread, if the thread has an ID.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>An async task.</returns>
+    /// <remarks>
+    /// In a service that hosts an agent, that is invoked via calls to the service, this might be at the end of each service call.
+    /// In a client application, this might be when the user closes the chat window or the application.
+    /// </remarks>
+    public async Task OnSuspendAsync(string? threadId, CancellationToken cancellationToken = default)
+    {
+        await Task.WhenAll(this.Behaviors.Select(x => x.OnSuspendAsync(threadId, cancellationToken)).ToList()).ConfigureAwait(false);
+    }
+
+    /// <summary>
+    /// Called when the current conversion is resumed and any state should be restored.
+    /// </summary>
+    /// <param name="threadId">The ID of the current thread, if the thread has an ID.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>An async task.</returns>
+    /// <remarks>
+    /// In a service that hosts an agent, that is invoked via calls to the service, this might be at the start of each service call where a previous conversation is being continued.
+    /// In a client application, this might be when the user re-opens the chat window to resume a conversation after having previously closed it.
+    /// </remarks>
+    public async Task OnResumeAsync(string? threadId, CancellationToken cancellationToken = default)
+    {
+        await Task.WhenAll(this.Behaviors.Select(x => x.OnResumeAsync(threadId, cancellationToken)).ToList()).ConfigureAwait(false);
```

Comment:
To list might not be necessary here, or can be simplified `ToArray`.
```suggestion
        await Task.WhenAll(this.Behaviors.Select(x => x.OnResumeAsync(threadId, cancellationToken))).ConfigureAwait(false);
```

Reply:
Thanks, updated.